schemas/components.schema.json

```diff
@@ -0,0 +1,39 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema",
+    "id": "https://immersive-web/gamepad-mapping/0.1.0/components.schema.json",
+    "title": "The components array",
+    "description": "The components array",
+    "type": "array",
+    "additionalItems": false,
+    "uniqueItems": true,
+    "minItems": 1,
+    "items": {
+        "title": "A component",
+        "description": "The joining of a data source and visualization information",
+        "type" : "object",
+        "properties": {
+            "dataSource" : {
+                "allOf": [ { "$ref": "mapping.schema.json#/definitions/mappingIndex"} ],
+                "description": "The index into the dataSources array"
+            },
+            "root" : {
+                "type": "string",
+                "description": "The name of the component's root node in the asset file"
+            },
+            "labelTransform" : {
+                "type": "string",
+                "description": "The name of the component's node in the asset file at which location a legend description can safely be placed"
+            },
+            "pressResponse" : {
+                "allOf": [ { "$ref": "mapping.schema.json#/definitions/mappingIndex"} ],
+                "description": "The index of the response to be invoked when the data source is pressed"
+            },
+            "touchResponse" : {
```

Comment:
Not entirely sure I'm following the comment...Are you suggesting that a single response element should have the nodes for both touch and press?

Reply:
I'm saying that the touch response element already does, so why does the differentiation at this level matter?

Reply:
oh!! Because it's not the same on a touchpad!  The touch response on a touchpad moves the little dot around.  the press response causes the whole touchpad to wobble.

Reply:
talked this through on the call.  will add a response type to each element in the responses array and have the component have a single responses id array.  rename responses to visualResponses